service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java

```diff
@@ -162,4 +166,48 @@ private void annotateDataWithQValue(List<String> studyIds, List<AlterationCountB
         return copyNumberCountByGenes;
     }
 
+    @Override
+    public List<GenericAssayDataCountItem> fetchGenericAssayDataCounts(List<String> molecularProfileIds, List<String> sampleIds,
+                                                               List<String> stableIds) throws MolecularProfileNotFoundException {
+        if (stableIds.isEmpty()) {
+            return new ArrayList<>();
+        }
+        
+        return genericAssayService.fetchGenericAssayData(molecularProfileIds, sampleIds, stableIds, "SUMMARY")
+            .stream()
+            .filter(g -> StringUtils.isNotEmpty(g.getValue()) && !g.getValue().equals("NA"))
+            .collect(Collectors.groupingBy(GenericAssayData::getGenericAssayStableId))
+            .entrySet()
+            .stream()
+            .map(entry -> {
+                int naCount = 0;
+                int totalCount = 0;
+                Map<String, List<GenericAssayData>> groupedByValue = entry.getValue().stream().collect(Collectors.groupingBy(GenericAssayData::getValue));
+                List<GenericAssayDataCount> counts = new ArrayList<>();
+                groupedByValue.forEach((value,dataList) -> {
+                    GenericAssayDataCount dataCount = new GenericAssayDataCount();
+                    dataCount.setValue(value);
+                    dataCount.setCount(dataList.size());
+                    counts.add(dataCount);
+                });
+
+                if (!counts.isEmpty()) {
+                    totalCount = counts.stream().mapToInt(GenericAssayDataCount::getCount).sum();
```

Comment:
isn't total count be calculated from `entry.getValue().size()` instead of using `counts`